lldb/include/lldb/Host/common/NativeProcessProtocol.h

```diff
@@ -301,7 +301,8 @@ class NativeProcessProtocol {
     return std::nullopt;
   };
 
-
+  // Handle exiting the GPU process when a native process exits.
+  virtual void HandleNativeProcessExit(const WaitStatus &exit_status) {};
```

Comment:
This function doesn't make sense for CPU processes.
Could you extend NativeProcessProtocol as a new class `GPUProcessProtocol` that has this additional method?
That would leave CPU process clean.
Then, make this function pure virtual

Reply:
I've changed my mind. You don't need this function at all. It's enough to add `NativeProcessDidExit` to `LLDBServerPlugin.h`. Each plugin should decide how they want to manage this high level event.